### PR TITLE
NEPT-2709: Restrict access to user_administrator

### PR DESCRIPTION
**Pull request by drishu**
### Description

Restrict access to user_administrator.

### Change log

- Added:
   - cce_basic_config_form_user_profile_form_alter()
   - cce_basic_config_query_alter

### Commands

drush cc all

### Checklist

- [x] Check if there are hook_updates and they are documented
- [ ] Add right labels to indicate in the devops ticket the commands to run
- [ ] Remove symlinks if it's a module removal

### CODE REVIEW

**Review by tomv-be — APPROVED** [review]

File: profiles/common/modules/features/cce_basic_config/cce_basic_config.module [hunk 1]
```diff
@@ -185,6 +185,17 @@ function cce_basic_config_form_alter(&$form, &$form_state, $form_id) {
   return $form;
 }
 
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
```
Comment (Fefaine) [inline profiles/common/modules/features/cce_basic_config/cce_basic_config.module:192]: We need some extra code comments here in order to remember later what we want to achieve

File: profiles/common/modules/features/cce_basic_config/cce_basic_config.module [hunk 2]
```diff
@@ -185,6 +185,17 @@ function cce_basic_config_form_alter(&$form, &$form_state, $form_id) {
...
+function cce_basic_config_form_user_profile_form_alter(&$form, &$form_state, $form_id) {
+  $account = user_load_by_name('user_administrator');
+
+  if (!empty($account) && $account->uid == $form['#user']->uid) {
+    drupal_goto('admin/people', [], 307);
```
Comment (Fefaine) [inline profiles/common/modules/features/cce_basic_config/cce_basic_config.module:198]: Ok, but super admin should be able to change password or account details of that user_administrator.

File: profiles/common/modules/features/cce_basic_config/cce_basic_config.module [hunk 3]
```diff
@@ -510,5 +521,26 @@ function cce_basic_config_js_alter(&$javascript) {
     'defer' => TRUE,
     'weight' => 5,
   ));
+}
 
+/**
+ * Implements hook_query_alter().
+ */
+function cce_basic_config_query_alter(QueryAlterableInterface $query) {
+  // Restrict access to user_administrator.
```
Comment (Fefaine) [inline profiles/common/modules/features/cce_basic_config/cce_basic_config.module]: Here also i would add a condition to not add this for uid 1
Imagine some site have deleted the role user_administrator

**Review by Fefaine — CHANGES_REQUESTED** [review]

File: profiles/common/modules/features/cce_basic_config/cce_basic_config.module (hunk 1 above)
Comment (drishu) [reply profiles/common/modules/features/cce_basic_config/cce_basic_config.module:192]: done, bit more below

File: profiles/common/modules/features/cce_basic_config/cce_basic_config.module (hunk 2 above)
Comment (drishu) [reply profiles/common/modules/features/cce_basic_config/cce_basic_config.module:198]: fixed

File: profiles/common/modules/features/cce_basic_config/cce_basic_config.module (hunk 3 above)
Comment (drishu) [reply profiles/common/modules/features/cce_basic_config/cce_basic_config.module]: fixed

**Pull request merged**